pkg/controllermanager/controller/cloudprofile/kubernetesversion/kubernetes_version.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetesversion
+
+import (
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	controllermgrconfig "github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/cloudprofile/helper"
+)
+
+// ReconcileKubernetesVersions reconciles the Spec.Kubernetes.Versions section of the CloudProfile
+func ReconcileKubernetesVersions(logger *logrus.Entry, config *controllermgrconfig.CloudProfileControllerConfiguration, cloudProfile *gardencorev1beta1.CloudProfile) (bool, *gardencorev1beta1.CloudProfile, error) {
+	if config == nil || config.KubernetesVersionManagement == nil || config.KubernetesVersionManagement.Enabled == false {
+		return false, cloudProfile, nil
+	}
+
+	profile := cloudProfile.DeepCopy()
+	supportedVersionsMap := make(map[string]helper.HighestSupportedVersion, len(profile.Spec.Kubernetes.Versions))
+	for cloudProfileIndex, currentVersion := range profile.Spec.Kubernetes.Versions {
```

Comment:
```suggestion
	for cloudProfileIndex, version := range profile.Spec.Kubernetes.Versions {
```
`currentVersion` is somehow missleading here